### PR TITLE
index_notation: implement the `divide` transformation

### PR DESCRIPTION
**Pull request by rohany**
The divide transformation divides a loop up into `n` equal components,
whereas split breaks a loop up into a components of size `n`.

### CODE REVIEW

**Comment by rohany** [comment]
This has a bug, don't merge it yet.

File: include/taco/index_notation/provenance_graph.h
Comment (stephenchouca) [inline include/taco/index_notation/provenance_graph.h:133]: This line looks a bit long. (There are a few other pretty long lines in this PR that should be broken up.)

File: test/tests-scheduling.cpp
Comment (stephenchouca) [inline test/tests-scheduling.cpp:888]: These test cases should be decomposed using `TEST_P`.

**Review by stephenchouca — CHANGES_REQUESTED** [review]
It would also be great if you can extend the command-line tool to support the divide transformation as well.

File: include/taco/index_notation/provenance_graph.h
Comment (rohany) [reply include/taco/index_notation/provenance_graph.h]: Done

File: test/tests-scheduling.cpp
Comment (rohany) [reply test/tests-scheduling.cpp:888]: I think that it is simpler for these parametrizations (that need to apply different scheduling statements which depend on the context) to be written this way than with `TEST_P`

**Comment by rohany** [comment]
> It would also be great if you can extend the command-line tool to support the divide transformation as well.

Done. Tested it locally, but once #393 lands I can add a CI test for it.

**Comment by Infinoid** [comment]
> Tested it locally, but once #393 lands I can add a CI test for it.

FYI, I've added a super basic CLI test for it.

https://github.com/tensor-compiler/taco/pull/393/files#diff-ed84bfc8bd6c1f68f013ec588a09bf94dd4ea4edb76f4a0b336294366e689634R177-R186

**Pull request merged**
